src/workerd/api/queue.h

```diff
@@ -108,24 +108,52 @@ struct IncomingQueueMessage {
   };
 };
 
+struct QueueRetryBatch {
+  bool value;
+  jsg::Optional<int> delaySeconds;
+  JSG_STRUCT(value, delaySeconds);
+};
+
+struct QueueRetryMessage {
+  kj::String msgId;
+  jsg::Optional<int> delaySeconds;
+  JSG_STRUCT(msgId, delaySeconds);
+};
+
 struct QueueResponse {
   uint16_t outcome;
+  QueueRetryBatch retryBatch;
+  // Deprecated, retryBatch should be used instead.
```

Comment:
I assume you'll come back to remove the old deprecated fields after the queues team's code that uses this has been updated?

Reply:
Yes, I am making these fields optional on the queue service side, and I will come back to remove them from here (in this PR).